### ISSUE TITLE
Is there a way to install and use Demucs without CUDA?

### ISSUE THREAD

**Issue by KaiVinter** [facebookresearch/demucs#395]
## ❓ Questions

I'm really in the need to do this because my phone is my main device right now, and I use Demucs a lot. I'm trying to install and run Demucs on Android via Termux, but since Android doesn't use an NVDIA GPU, CUDA does not work/install. But I would like to use it only with the CPU. Is there a way to do this?

**Comment by KaiVinter** [comment]
If needed: I'm not training anything, I just use it as a musician to separate audio.

**Comment by dillfrescott** [comment]
Yes, I've been using it for months without cuda. It should default to cpu.

**Comment by dillfrescott** [comment]
Also I would recommend the new version 4! Just download the `ht` branch code, make sure you have python installed, and just run `pip instal -e .` from the ht folder. That should install it for you!

**Comment by GUUser91** [comment]
Here's how to install the bleeding edge version of demucs on termux.
First install python and pip
`pkg install python python-pip`

Then install these dependencies
`pkg install python-lameenc python-torch python-torchaudio`

Now install demucs
`pip install -U git+https://github.com/facebookresearch/demucs#egg=demucs`

**Comment by AARP41298** [comment]
if
```
ImportError: dlopen failed: cannot locate symbol "_ZN4absl12lts_2023080213hash_internal15MixingHashState5kSeedE" referenced by "/data/data/com.termux/files/usr/lib/python3.11/site-packages/torch/lib/libtorch_python.so"
```
then
https://github.com/termux/termux-packages/issues/20158#issuecomment-2113284083
